File: Daily Flash/week 4/day 1/18-DailyFlash_Solutions/3-Feb_Solutions_Three/C/prog1.c
#include<stdio.h>

int isPrime(int n) {
	if(n <= 3) {
		return n > 1;
	} else if(n % 2 == 0 || n % 3 == 0) {
		return 0;
	}

	int i = 5;

	while(i * i <= n) {
		if(n % i == 0 || n % (i + 2) == 0) {
			return 0;
		}	
		i += 6;
	}

	return 1;
}	

int main(void) {
	int num;
	printf("Enter number : ");
	scanf("%d", &num);
	if(isPrime(num)) {
		printf("Prime number.");
	} else {
		printf("Not a prime");
	}
	return 0;
}	

File: Daily Flash/week 4/day 1/18-DailyFlash_Solutions/3-Feb_Solutions_Three/Python/prog1.py
def isPrime(num):
    if(num <= 3):
        return num > 1
    elif(num % 2 == 0 or num % 3 == 0):
        return False

    i = 5

    while(i * i <= num):
        if(num % i == 0 or num % (i + 2) == 0):
            return False
        i += 6

    return True

num = int(input("num : "))
if(isPrime(num)):
    print("Prime.")
else:
    print("Not prime.");
